staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -314,3 +317,41 @@ func (e *eventBroadcasterImpl) StartRecordingToSink(stopCh <-chan struct{}) {
 		stopWatcher()
 	}()
 }
+
+type eventClientImpl struct {
+	CoreClient         typedv1core.EventsGetter
+	CoreBroadcaster    record.EventBroadcaster
+	V1beta1Client      typedv1beta1.EventsGetter
+	V1beta1Broadcaster EventBroadcaster
+}
+
+// NewEventClient Creates a new event broadcaster.
+func NewEventClient(client clientset.Interface) *EventClient {
```

Comment:
can we add another commit for refactoring the scheduler to use what we added ?

Reply:
We should. Though I explicitly asked Chelsey to start with simple WIP PR to show how it would look like and what dependencies appear etc. to confirm that it actually make sense.

Reply:
ack